test/Prompt/DontWrap.C

```diff
@@ -172,7 +172,7 @@ const Test2::Nested::Value::type & Test2::Nested::reference() const {
 Test2::Nested Nest2(80);
 // CHECK: Nested::Nested(160)
 Nest2.simpleAdd(3)
-// CHECK: (Test2::Nested::Value::type) 83
+// CHECK: (int) 83
```

Comment:
[cling] std::vector<int> a(5,5);
[cling] a.begin()
[cling] a\[0\]
A. (std::__1::vector\<int, std::__1::allocator\<int\> \>::iterator) @0x7fd76077e480
a. (std::__1::__vector_base\<int, std::__1::allocator\<int\> \>::value_type) 5
B. (std::__1::__wrap_iter\<int *\>) @0x7f8340653d30
b. (int) 5
C. (std::vector\<int, std::allocator\<int\> \>::iterator) @0x7fad4cf7bbf0
c. (int) 5

So what should happen?
A is behavior before the PR,
B is behavior at PR
C is updating the PR to reflect changes discussed